lms/services/assignment.py

```diff
@@ -62,6 +71,40 @@ def upsert_assignment(
 
         return assignment
 
+    def upsert_assignment_membership(
+        self, assignment: Assignment, lti_user: LTIUser
```

Comment:
If we take `user` here we could do `user.roles` now but as we remove that we should take `roles` separaptly.

I think that would make sense, the user is not tied to a role so it makes sense it's two separate concepts arguments.

Reply:
Eugh this is a bit of a mess. Presumably `lti_user` just gets roles from the params anyway, so it might be clearer to totally ignore it.

We don't want to get mixed up in `user.roles` if we are going to get rid of them, so I think separate args might be the way.